### PR TITLE
Bitcoin backports #12763 #17511 #17675 #17617 #17698 #17703 #17705 #17787 #17806 #17743

### PR DESCRIPTION
**Pull request by knst**
Need to test on mac OS: bitcoin#17787

By some reason functional test rpc_whitelist.py doesn't work with option --usecli, in bitcoin's original code I don't see a flag "skip cli" for this test.

### CODE REVIEW

File: configure.ac [hunk 1]
```diff
@@ -1266,6 +1266,29 @@ else
     fi
 fi
 
+AC_MSG_CHECKING([for std::system])
```
Comment (kittywhiskers) [inline configure.ac:1269]: Already present in https://github.com/dashpay/dash/pull/4864 as https://github.com/dashpay/dash/pull/4864/commits/6627bc968873a86e46c2eb72f64a0d30c9c32f11.

Comment (knst) [reply configure.ac]: removed

**Comment by knst** [comment]
I think [56a9f9b] should be backported to 18.x also

**Review by UdjinM6 — DISMISSED** [review]
utACK

File: src/httprpc.cpp [hunk 2]
```diff
@@ -15,7 +15,12 @@
 #include <util/translation.h>
 #include <walletinitinterface.h>
 
+#include <algorithm>
```
Comment (PastaPastaPasta) [inline src/httprpc.cpp:18]: 12763 needs some release notes

**Review by PastaPastaPasta — CHANGES_REQUESTED** [review]

**Comment by PastaPastaPasta** [comment]
(otherwise LGTM, simply add a release note in doc/ that describes the changes that 12763 includes)

**Review by PastaPastaPasta — APPROVED** [review]
utACK for merging via merge commit

**Review by UdjinM6 — APPROVED** [review]
re-utACK

**Pull request merged**